util/grpc/src/lib.rs

```diff
@@ -67,21 +67,26 @@ pub fn send_result<T>(
 ) {
     let logger = logger.clone();
     let success = resp.is_ok();
+    let mut code = RpcStatusCode::OK;
 
     match resp {
         Ok(ok) => ctx.spawn(
             sink.success(ok)
                 .map_err(move |err| log::error!(logger, "failed to reply: {}", err))
                 .map(|_| ()),
         ),
-        Err(e) => ctx.spawn(
-            sink.fail(e)
-                .map_err(move |err| log::error!(logger, "failed to reply: {}", err))
-                .map(|_| ()),
-        ),
+        Err(e) => {
+            code = e.code();
+            ctx.spawn(
+                sink.fail(e)
+                    .map_err(move |err| log::error!(logger, "failed to reply: {}", err))
+                    .map(|_| ()),
+            )
+        }
     }
```

Comment:
This should be `let code = match { .. }`

Reply:
I don't think that's the case. The issue referenced specified that a list of gRPC status codes for FOG services should be tracked. All the above code does is record it so that it can be tracked by a single gRPC function below. I'm not sure what the match would achieve.

The surrounding function `send_result` is called by most FOG gRPC methods to do the actual work of spawning a task into the gRPC async context. Changing the match in `send_result` to match on error codes vs. the `resp` Result would significantly changing the way FOG services currently interact with gRPC.

Reply:
We can avoid `mut` with a second match
```suggestion
    let code = match &resp {
        Ok(_) => RpcStatusCode::OK,
        Err(e) => e.code(),
    };
    match resp {
        Ok(ok) => ctx.spawn(
            sink.success(ok)
                .map_err(move |err| log::error!(logger, "failed to reply: {}", err))
                .map(|_| ()),
        ),
        Err(e) => ctx.spawn(
            sink.fail(e)
                .map_err(move |err| log::error!(logger, "failed to reply: {}", err))
                .map(|_| ()),
        ),
    }
```

Reply:
Ah yes, I see the thought